client/jetpack-connect/controller.js

```diff
@@ -204,19 +228,17 @@ export function signupForm( context, next ) {
 	removeSidebar( context );
 
 	const { query } = context;
-	const validQueryObject = validator( authorizeQueryDataSchema )( query );
-
-	if ( validQueryObject ) {
-		const transformedQuery = authQueryTransformer( query );
+	const transformedQuery = parseAuthorizationQuery( query );
+	if ( transformedQuery ) {
 		context.store.dispatch( startAuthorizeStep( transformedQuery.clientId ) );
 
-		let interval = context.params.interval;
-		let locale = context.params.locale;
-		if ( context.params.localeOrInterval ) {
-			if ( [ 'monthly', 'yearly' ].indexOf( context.params.localeOrInterval ) >= 0 ) {
-				interval = context.params.localeOrInterval;
+		let { interval, locale } = context.params;
+		const { localeOrInterval } = context.params;
+		if ( localeOrInterval ) {
+			if ( includes( [ 'monthly', 'yearly' ], localeOrInterval ) ) {
+				interval = localeOrInterval;
```

Comment:
I'll roll back the changes in this section. They're out of scope, and interval appears to be completely irrelevant for signup/authorize. I'll handle that in another PR.